### PR TITLE
NO-TICKET - Fix sending empty alarms for send

### PR DESCRIPTION
**Pull request by gslevinkov**
# THIS MR IS OPTIONAL FOR MERGE, THERE NO BUGFIX TICKET FOR THIS

## Description of Work

- fix sending empty body when formatting throwable class

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/alarmlogger/AlarmErrorsModule.kt [hunk 1]
```diff
@@ -24,10 +29,31 @@ object AlarmErrorsModule : InjectionModule {
             )
             retrofit.create()
         }
-        factoryOf(::AlarmErrorsLogger)
+        factory {
+            // check the children size of  AlarmFeatureConverter to check DI is valid
+            val converters: List<AlarmFeatureConverter> = listOf(
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/alarmlogger/AlarmErrorsModule.kt:34]: do not be alarms, this is for scalability 

`AlarmFeatureConverter` - is a base interface for all the alarm converters. We already have 7 of them and more will come!

`List<AlarmFeatureConverter>` comes as a field to root logger and has a method that allows to get typed `T: AlarmFeatureConverter` for the requests

File: app/src/main/java/org/p2p/wallet/alarmlogger/model/AlarmBridgeErrorConverter.kt [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+import org.p2p.wallet.alarmlogger.api.AlarmErrorsBridgeClaimRequest
+import org.p2p.wallet.alarmlogger.api.AlarmErrorsRequest
+import org.p2p.wallet.alarmlogger.api.AlarmErrorsSendBridgeRequest
+
+class AlarmBridgeErrorConverter(
+    private val gson: Gson,
+    private val throwableFormatter: AlarmThrowableFormatter
+) : AlarmFeatureConverter {
+
+    fun toBridgeClaimErrorRequest(
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/alarmlogger/model/AlarmBridgeErrorConverter.kt:17]: bridge claim errors extracted to a separate converter

File: app/src/main/java/org/p2p/wallet/alarmlogger/model/AlarmSendErrorConverter.kt [hunk 3]
```diff
@@ -235,36 +168,4 @@ class AlarmSendErrorConverter(
...
-    is SimulationException -> "Simulation error: ${getDirectMessage().orEmpty()}"
-    else -> emptyString()
-}
-
-private fun Throwable.getFeeRelayerError(): String = when (this) {
-    is FeeRelayerException -> "FeeRelayer error: ${getDirectMessage().orEmpty()}"
-    else -> emptyString()
-}
-
-private fun Throwable.getBlockchainError(): String = when (this) {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/alarmlogger/model/AlarmSendErrorConverter.kt:262]: this formatting logic is extracted to a separate throwable formatter special for such cases

File: app/src/main/java/org/p2p/wallet/alarmlogger/model/AlarmThrowableFormatter.kt [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+import java.net.UnknownHostException
+import org.p2p.core.network.data.ServerException
+import org.p2p.core.network.data.SimulationException
+import org.p2p.wallet.bridge.model.BridgeResult
+import org.p2p.wallet.feerelayer.model.FeeRelayerException
+import org.p2p.wallet.newsend.model.SendTransactionFailed
+
+class AlarmThrowableFormatter {
+    private fun Throwable.unwrap(): Throwable = if (this is SendTransactionFailed) cause else this
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/alarmlogger/model/AlarmThrowableFormatter.kt:11]: the great fix

**Review by 1sakov — APPROVED** [review]

**Pull request merged**